blogpost-drools-model-compiler.md

```diff
@@ -0,0 +1,116 @@
+
+## Method #2: create a self contained project
+
+Let's use the Drools archetype and verify that your small reproducer is working against Drools Legacy. 
+
+Start by creating a KJAR using the [Maven Archetype](https://mvnrepository.com/artifact/org.kie/kie-drools-archetype), modify the rules and the test to verify everything is working accordingly. The default archetype will run the test against the DRL mode.
+
+Change the generated model to build an executable model KJAR. To do so, switch in the `pom.xml` from `drools-engine-classic` to `drools-engine`. Also add the `drools-model-compiler` dependency.
+
+Compile the project using maven and the command line. Use 
+
+`mvn clean install -DskipTests=true` 
+
+as it'll try run the tests using the classic engine but we don't have the `drools-mvel` dependency in the class path anymore.
+
+![kjar](kjar.gif)
+
+Verify the Executable Model has been built in the KJAR, you can for example using the command
+
+`jar -tf target/name_of_the_kjar.jar`
+
+And you'll see all the Executable Model classes under and the `drools-model` file. 
+You can also check the `target` subfolder of maven and you'll see the generated files.
+Yet another way to do it is to check the maven log for this phrase:
+
+```
+
+INFO] Found 7 generated files in Canonical Model
+[INFO] Generating /Users/lmolteni/git/contribute/reproducer-kjar/target/generated-sources/drools-model-compiler/main/java/./org/example/P41/LambdaExtractor41A2683D222972683028514525A5437B.java
+...
+
+```
+
+Create another project called `runner` that has the original project as a dependency in the Maven `pom.xml`. 
+The same archetype can be used again, but you have to change a few things
+
+- Remove all the classes in `src/main/java`
+- Remove the DRL files, as this project only consumes KJARs
+- Remove the `kmodule.xml` file  
+- Remove the `kie-maven-plugin` from the build. Again, we don't want to build KJARs here, only to use them.
+- Switch in the `pom.xml` from `drools-engine-classic` to `drools-engine`.
+- Remove the `kjar` packaging in pom.xml   
+- Add the original KJAR dependency
+
+```xml
+    <dependency>
+      <groupId>org.example</groupId>
+      <artifactId>reproducer-kjar</artifactId>
+      <version>1.0-SNAPSHOT</version>
+    </dependency>
+```
+
+- Move the test from the original reproducer to this module
+- Never call `kContainer.verify();` while using Executable Model KJARs as this will retrigger the build
+
+
+Run the test. This time the test will run using the Executable Model.
+
+![runner](runner.gif)
+
+You can see it because there will be this line in the logs
+
+```
+2021-06-15 11:32:29,576 INFO  [org.drools.modelcompiler.CanonicalKieModuleProvider] (main) Artifact org.example:reproducer-kjar:1.0-SNAPSHOT has executable model
+```
+
+This can probably be abstracted in a new archetype, let us know if you're interested and we can work of it. 
```

Comment:
I agree that it should be nice to create an archetype that already produces both kjar module and the runner